app/controllers/shipments_controller.rb

```diff
@@ -0,0 +1,68 @@
+require "timeout"
+class ShipmentsController < ApplicationController
+
+  skip_before_filter :verify_authenticity_token, only: :ship
+
+  def index
+    render json: { errors: "This page does not exist. Perhaps you were looking for 'adaships.herokuapp.com/log' or 'adaships.herokuapp.com/ship'."}, status: 404
+  end
+
+  def ship
+    @shipping_data = JSON.parse(request.body.read)
+
+    if  @shipping_data.keys.length != 5 ||
+        @shipping_data.values.select(&:nil?).length != 0 ||
+        @shipping_data.values.select(&:empty?).any?
+      render json: { errors: "Incomplete request."}, status: :bad_request
```

Comment:
If I were a consumer of this API I'd be frustrated by this message that doesn't tell me _what_ is incomplete about the request.